wordpress-mobile/WordPress-Android
# Fix/reader blog table

This PR fixes an issue with identifying a feed before a post makes it to the `ReaderSiteHeaderView` code.

Modifications include

- A new static method was added to `ReaderUtils` to determine if this is a feed and uses this value to make decisions in the header code. 
- When parsing data for blogtable, make sure to pull blogId in from response
- When requesting to follow a site from the header, don't send down the path to display enable notifications when this is a feed.
- Updated logic in `ReaderActivityLauncher` to check if this is a feed before adding feedId or blogId to arguments 
- Removed old code associated with previous [Update Reader Header Site UI ](https://github.com/wordpress-mobile/WordPress-Android/pull/12655)
- Removes animation from Following button

**Open question**: Should `ReaderUtils.isFeed` be renamed to `isExternalFeed`?

To test:
Test A: validate follow and enable subscription works for a site
- Log into the app
- Search for a site
- Tap the post header line to bring up the post card
- Tap the follow button in the header & quickly tap on the ENABLE notifications text in the snackbar
- Leave the app open because you'll need it again
- Validate that you are subscribed to the site you selected (Use your Wordpress.com site or rest client to run `/read/following/mine?meta=site,feed`)
- Tap the unfollow button for the same site
- Validate that the subscription for the site is gone
- Log out of the app

Test B: Validate that you are not prompted to enable notifications on tap of follow from header
- Log into the app
- Go to reader -> gear -> 
- Enter an RSS feed to follow: https://timesofindia.indiatimes.com/rssfeedstopstories.cms & tap the follow icon
- Back out to the reader and notice the feed shows up in following
- Tap on the header line of the post card
- The new Site header is shown as "Following"
- Tap the "following" button to unfollow
- The following button turns pink
- Tap to Follow 
- Note that you are not prompted to enable subscriptions - no snackbar is visible


PR submission checklist:
